shimming-toolbox/test/cli/test_cli_b0shim.py

```diff
@@ -115,18 +120,32 @@ def test_cli_dynamic_sph(self, nii_fmap, nii_target, nii_mask, fm_data, target_d
 
             assert values == [0.002985, -14.587414, -57.016499, -2.745062, -0.401786, -3.580623, 0.668977, -0.105560]
 
-    def test_cli_dynamic_bfgs(self, nii_fmap, nii_target, nii_mask, fm_data, target_data):
+            res_soft = runner.invoke(b0shim_cli, ['dynamic',
+                                             '--fmap', fname_fmap,
+                                             '--target', fname_target,
+                                             '--mask', fname_softmask,
+                                             '--scanner-coil-order', '1,2',
+                                             '--regularization-factor', '0.1',
+                                             '--slices', 'ascending',
+                                             '--output', tmp],
+                                catch_exceptions=False)
+
+            assert res_soft.exit_code == 0
+
+    def test_cli_dynamic_bfgs(self, nii_fmap, nii_target, nii_mask, nii_softmask, fm_data, target_data):
```

Comment:
We should aim to testing 1 thing per test. We should test that optimization with soft masks works using the CLI, but we should probably test all the different residual functions in test_optimizer rather than testing the whole pipeline in here.